op-chain-ops/genesis/setters.go

```diff
@@ -89,3 +89,20 @@ func setupPredeploy(db vm.StateDB, deployResults immutables.DeploymentResults, s
 
 	return nil
 }
+
+// setup4788Contract creates the EIP-4788 beacon-block-roots contract, part of the Ecotone upgrade.
+func setup4788Contract(db vm.StateDB) error {
+	// EIP-4788 defines a deterministic deployment transaction that deploys the beacon-block-roots contract.
+	// To embed the contract in genesis, we want the deployment-result, not the contract-creation tx input code.
+	// Since the contract deployment result is deterministic and the same across every chain,
+	// the bytecode can be easily verified by comparing it with chains like Goerli.
+	// During deployment it does not modify any contract storage, the storage starts empty.
+	// See https://goerli.etherscan.io/tx/0xdf52c2d3bbe38820fff7b5eaab3db1b91f8e1412b56497d88388fb5d4ea1fde0
+	// And https://eips.ethereum.org/EIPS/eip-4788
+	eip4788ContractAddr := common.HexToAddress("0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02")
+	eip4788ContractCode := common.Hex2Bytes("0x3373fffffffffffffffffffffffffffffffffffffffe14604d57602036146024575f5ffd5b5f35801560495762001fff810690815414603c575f5ffd5b62001fff01545f5260205ff35b5f5ffd5b62001fff42064281555f359062001fff015500")
+	db.CreateAccount(eip4788ContractAddr)
+	db.SetCode(eip4788ContractAddr, eip4788ContractCode)
+	db.SetNonce(eip4788ContractAddr, 1) // TODO review why goerli nonce is 1
```

Comment:
Was this already deployed on goerli? This feels like a pretty important todo to figure out.

Reply:
This is expected behavior for it to be 1, because you can deploy no code it increments the nonce of the account. see https://github.com/ethereum/go-ethereum/blob/daa2e5d6a66833b9834b60a3a46835610bbde99a/core/vm/evm.go#L450